perun-tasks-lib/src/main/java/cz/metacentrum/perun/taskslib/dao/jdbc/TaskDaoJdbc.java

```diff
@@ -176,7 +180,7 @@ public Task getTask(int execServiceId, int facilityId, int engineID) {
 						", " + ExecServiceDaoJdbc.execServiceMappingSelectQuery + ", " + ServicesManagerImpl.serviceMappingSelectQuery + " from tasks left join exec_services on tasks.exec_service_id = exec_services.id " +
 						"left join facilities on facilities.id = tasks.facility_id left join services on services.id = exec_services.service_id " +
 						"where exec_services.id = ? and tasks.facility_id = ? and tasks.engine_id = ?",
-						new Integer[] { execServiceId, facilityId, engineID }, TASK_ROWMAPPER);
+						new Object[] { execServiceId, facilityId, engineID < 0 ? "null" : engineID}, TASK_ROWMAPPER);
```

Comment:
Regarding the fact, that you can't store string in integer column, you should modify select instead to distinct between ``tasks.engine_id is null`` and ``tasks.engine_id = "+engineId+"``. I now joining strings is not generally safe, but here value is a generic ``int``, which is harmless regarding sql injection.